src/browser/base/content/ZenUIManager.mjs

```diff
@@ -583,4 +585,95 @@ var gZenVerticalTabsManager = {
     }
     target.appendChild(child);
   },
+
+  //_insertDoubleClickListenerPinnedTabs() {
+  //  const tabs = gBrowser.tabs;
+  //  for (const tab of tabs) {
+  //    tab.addEventListener('dblclick', this.renameTabStart.bind(this));
+  //  }
+  //},
+
+  renameTabKeydown(event) {
+    if (event.key === 'Enter') {
+      let label = this._tabEdited.querySelector('.tab-label-container-editing');
+      let input = this._tabEdited.querySelector('#tab-label-input');
+      let newName = input.value.trim();
+
+      // Check if name is blank, reset if so
+      if (newName) {
+        this._tabEdited.label = newName;
+        this._tabEdited.setAttribute('zen-has-static-label', 'true');
+      } else {
+        // If the page is loaded, get the title of the page. Otherwise, keep name as is
+        this._tabEdited.label = gBrowser.getBrowserForTab(this._tabEdited).contentTitle || this._tabEdited.label;
+        // If the page had a title, reset the zen-has-static-label attribute
+        if (gBrowser.getBrowserForTab(this._tabEdited).contentTitle) {
```

Comment:
Every page has a title, id just remove this if-statement

Reply:
Pages that don't have `document.title` generally use the URL as the page label in the tab bar. Does the browser capture this under `contentTitle`?